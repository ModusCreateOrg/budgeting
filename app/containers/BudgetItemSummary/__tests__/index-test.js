import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Default, { BudgetItemSummary } from '../';

jest.mock('components/BudgetItemHeader', () => () => <div>BudgetItemHeader</div>);
jest.mock('components/PieChart', () => () => <div>PieChart</div>);

it('renders children correctly for outflow transaction', () => {
  const outflowTransaction = {
    id: 1,
    description: 'outflow transaction',
    value: -4500,
    categoryId: '16',
  };
  const outFlowBalance = 15000;
  const elem = renderer
    .create(<BudgetItemSummary transaction={outflowTransaction} balance={outFlowBalance} />)
    .toJSON();
  expect(elem).toMatchSnapshot();
});

it('renders children correctly for inflow transaction', () => {
  const inflowTransaction = {
    id: 1,
    description: 'inflow transaction',
    value: 5000,
    categoryId: '16',
  };
  const inflowBalance = 32000;

  const elem = renderer.create(<BudgetItemSummary transaction={inflowTransaction} balance={inflowBalance} />).toJSON();
  expect(elem).toMatchSnapshot();
});

it("dispatches 'goBack' callback when back button is clicked", () => {
  const inflowTransaction = {
    id: 1,
    description: 'inflow transaction',
    value: 5000,
    categoryId: '16',
  };
  const inflowBalance = 32000;

  const mockCallBack = jest.fn();

  const elem = shallow(
    <BudgetItemSummary transaction={inflowTransaction} balance={inflowBalance} goBack={mockCallBack} />
  );
  elem.find('button').simulate('click');
  expect(mockCallBack).toBeCalled();
});
