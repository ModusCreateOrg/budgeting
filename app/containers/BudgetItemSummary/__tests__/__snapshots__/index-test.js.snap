// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders children correctly for inflow transaction 1`] = `
<section
  className="budgetItemSummary"
>
  <button
    onClick={undefined}
  >
    Back
  </button>
  <div>
    BudgetItemHeader
  </div>
  <div>
    PieChart
  </div>
  ;
</section>
`;

exports[`renders children correctly for outflow transaction 1`] = `
<section
  className="budgetItemSummary"
>
  <button
    onClick={undefined}
  >
    Back
  </button>
  <div>
    BudgetItemHeader
  </div>
  <div>
    PieChart
  </div>
  ;
</section>
`;
